config/interpolate_funcs.go

```diff
@@ -54,12 +54,13 @@ func interpolationFuncConcat() ast.Function {
 					continue
 				}
 
-				if strings.Contains(argument, InterpSplitDelim) {
+				if IsStringList(argument) {
 					isDeprecated = false
+					finalList = append(finalList, StringList(argument).Slice()...)
+				} else {
+					finalList = append(finalList, argument)
 				}
 
-				finalList = append(finalList, argument)
```

Comment:
This was implicitly relying on the fact that the old representation of StringLists allowed them to be arbitrarily combined with bare elements.

For example, previously: `["foo", "barSLDbaz", "qux"]` would make it through the `strings.Join` as a valid list. Now we need to respect the encapsulation of the type and handle it explicitly.